### PR TITLE
Finish show activities feature

### PR DESCRIPTION
**Pull request by ttuan**

### CODE REVIEW

File: app/models/lesson.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
 class Lesson < ActiveRecord::Base
+  included ActivityLog
+
   belongs_to :user
   belongs_to :category
   has_many :results, dependent: :destroy
+
+  private
+  def create_learned_activity
+    create_activity user_id, id, "learned"
```
Comment (ThuBM) [inline app/models/lesson.rb]: move "learned" to Settings.

File: app/views/activities/_activity.html.erb [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
-</ol>
+<div class="row">
+  <div class="col-md-9 col-md-offset-2">
+    <hr>
+    <h4>
+      <% type = activity.type %>
+      <%= gravatar_for activity.user %>
+      <b><%= link_to activity.user.name, activity.user %></b>
+      <%= activity.status %>
+      <% if activity.status == "learned" %>
```
Comment (ThuBM) [inline app/views/activities/_activity.html.erb]: move "learned" to Settings

File: app/models/relationship.rb [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+
   validates :follower_id, presence: true
   validates :followed_id, presence: true
+
+  after_save :create_follow_activity
+  before_destroy :create_unfollow_activity
+
+  private
+  def create_follow_activity
+    create_activity follower_id, followed_id, "follow"
```
Comment (ThuBM) [inline app/models/relationship.rb]: move "follow" to Settings

File: app/models/relationship.rb [hunk 4]
```diff
@@ -1,6 +1,21 @@
...
+  end
+
+  def create_unfollow_activity
+    create_activity follower_id, followed_id, "unfollow"
```
Comment (ThuBM) [inline app/models/relationship.rb]: move "unfollow" to Settings

File: app/views/users/_unfollow.html.erb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <%= form_for current_user.active_relationships.find_by(followed_id: @user.id),
              html: {method: :delete},
```
Comment (ThuBM) [inline app/views/users/_unfollow.html.erb]: indent

File: app/models/activity.rb [hunk 6]
```diff
@@ -1,4 +1,15 @@
 class Activity < ActiveRecord::Base
   belongs_to :user
+  scope :feed_all, ->user{where "user_id IN (?) OR user_id = ?", user.following_ids, user.id}
+  scope :feed_own, ->user{where "user_id = ?", user.id}
+  scope :recent, ->{order "created_at DESC"}
+
+  enum status: [:learned, :follow, :unfollow]
```
Comment (ThuBM) [inline app/models/activity.rb]: do you think this status is necessary?

File: app/views/users/show.html.erb [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
-      <%= render "follow_form" if logged_in? %>
-    </p>  
-  </aside>
+  <div class="col-md-3">
+    <aside>
+      <section class="user_info">
+        <%= link_to gravatar_for(@user, size: Settings.image_size100), @user %>
+      </section>
+      <p>
+        <%= render "follow_form" if logged_in? %>
```
Comment (ThuBM) [inline app/views/users/show.html.erb]: 1. users can follow themself ?? 
2. if user followed an other, still rendered follow form??

File: config/settings.yml [hunk 8]
```diff
@@ -20,3 +20,6 @@ filter:
   all: "get_all"
   learned: "learned"
   not_learned: "not_learned"
+follow: follow
+unfollow: unfollow
+learned: learned
```
Comment (ThuBM) [inline config/settings.yml]: new line

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -8,6 +8,8 @@ def index
   end
 
   def show
+    @feed_activities = Activity.feed_own(@user).recent.paginate page: params[:page],
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: @user.activities

File: app/models/activity.rb [hunk 10]
```diff
@@ -1,4 +1,14 @@
 class Activity < ActiveRecord::Base
   belongs_to :user
-end
+  scope :feed_all, ->user{where "user_id IN (?) OR user_id = ?", user.following_ids, user.id}
```
Comment (chuanhtuan) [inline app/models/activity.rb]: đoạn code này sẽ chạy rất chậm vì phải run 2 query
(1 cho user.following_ids)

File: app/controllers/static_pages_controller.rb [hunk 11]
```diff
@@ -1,8 +1,8 @@
 class StaticPagesController < ApplicationController
   def home
     if logged_in?
-      @feed_activities = current_user.feed.paginate page: params[:page],
-        per_page: Settings.per_page
+      @feed_activities = Activity.feed_all(current_user.following_ids, current_user).recent
+      .paginate page: params[:page], per_page: Settings.per_page
```
Comment (ThuBM) [inline app/controllers/static_pages_controller.rb]: indent

**Pull request merged**